scripts/generate_dump

```diff
@@ -1511,9 +1529,49 @@ while getopts ":xnvhzas:t:r:d" opt; do
             ;;
         /?)
             echo "Invalid option: -$OPTARG" >&2
-            exit 1
+            exit $EXT_GENERAL
             ;;
     esac
 done
 
-main
+# Check permissions before proceeding further
+if [ `whoami` != root ] && ! $NOOP;
+then
+    echo "$0: must be run as root (or in sudo)" >&2
+    exit 10
+fi
+
+##
+## Attempt Locking
+##
+
+if mkdir "${LOCKDIR}" &>/dev/null; then
+    trap 'handle_exit' EXIT
+    echo "$$" > "${PIDFILE}"
+    # This handler will exit the script upon receiving these interrupts
+    # Trap configured on EXIT will be triggered by the exit from handle_signal function
+    trap 'handle_signal' SIGINT SIGHUP SIGQUIT SIGTERM
+    echo "Lock Succesfully Accquired and Installed signal handlers"
+    # Proceed with the actual code
+    main
+else
+    # lock failed, check if the other PID is alive
+    PID_PROG="$(cat "${PIDFILE}")"
+
+    if [ $? != 0 ]; then
+        # Another instance is probably about to remove the lock or PIDfile doesn't exist
+        # Wait for a few secs and restart
+        sleep "$(( $RANDOM % 2 + 1)).$(( $RANDOM % 999 ))"
+        rm_lock_and_restart
```

Comment:
It's easy to introduce bug if restarting. Just exit and let user make decision.

Reply:
This can automatically be invoked as well from auto-techsupport feature. 

Also, If the logic reaches this point, it means that the directory exist but there is some problem with reading the PID file (probably because the other instance is removing it). Ref: https://wiki.bash-hackers.org/howto/mutex & http://wiki.grzegorz.wierzowiecki.pl/code:mutex-in-bash.

So if that's the case, why give up the opportunity to collect the dump.

Reply:
Handled. Can you check and verify